engine/runtime-benchmarks/src/main/java/org/enso/interpreter/bench/benchmarks/semantic/MultiValueBenchmarks.java

```diff
@@ -0,0 +1,155 @@
+package org.enso.interpreter.bench.benchmarks.semantic;
+
+import java.util.concurrent.TimeUnit;
+import org.enso.common.MethodNames;
+import org.enso.compiler.benchmarks.Utils;
+import org.graalvm.polyglot.Value;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.BenchmarkParams;
+import org.openjdk.jmh.infra.Blackhole;
+
+/**
+ * These benchmarks compare performance of {@link EnsoMultiValue}. They create a vector in a certain
+ * configuration representing numbers and then they perform {@code sum} operation on it.
```

Comment:
@Akirathan, @hubertp, @4e6 (and also @radeusgd):
- these are the benchmarks for _intersection types_
- a `Vector` is created by `make_vector typ` method
- various `typ`es include `Integer`, `Float`, `Complex` as non-intersection type _benchmarks to compare to_
- then there are various `typ`es mixing `Float` and `Complex` together into different _intersection types_

Ideally the _intersection types_ benchmark results should be close to [base benchmark](https://github.com/enso-org/enso/pull/11924/files#r1894574140) `sumOverComplexBaseBenchmark0` with some _"reasonable overhead"_.

Are you OK with these benchmarks?